### PR TITLE
TMDM-12982 : Daikon TQL integration in UserQueryBuilder for where clauses

### PR DESCRIPTION
**Pull request by YohannTalend**
https://jira.talendforge.org/browse/TMDM-12982

**What is the current behavior?** (You should also link to an open issue here)

No support TQL in UserQueryBuilder

**What is the new behavior?**

Integration Daikon TQL in UserQueryBuilder for many operators : 

- Comparator : EQ, LT, GT, NED, LET, GET
- IsEmpty
- Contains
- Not
- In

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.query/src/org/talend/mdm/query/BasicConditionProcessor.java [hunk 1]
```diff
@@ -49,7 +49,12 @@ public Condition process(JsonObject condition, MetadataRepository repository) {
                     throw new IllegalArgumentException("Value '" + valueElement + "' is not supported."); //$NON-NLS-1 //$NON-NLS-2
                 }
             } else {
-                expression = Deserializer.getTypedExpression(element.getAsJsonObject()).process(element.getAsJsonObject(), repository);
+                final TypedExpression process = Deserializer.getTypedExpression(element.getAsJsonObject()).process(element.getAsJsonObject(), repository);
```
Comment (lpw82gf) [inline org.talend.mdm.query/src/org/talend/mdm/query/BasicConditionProcessor.java]: What's the usage for this change?

File: org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java [hunk 2]
```diff
@@ -42,6 +45,10 @@ private UserQueryBuilder(Expression expression) {
         this.expression = expression;
```
Comment (hw-zhu) [inline org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java]: Pls update Copyright information for related class.

File: org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java [hunk 3]
```diff
@@ -854,33 +861,43 @@ public UserQueryBuilder select(TypedExpression expression) {
...
             throw new IllegalArgumentException("Condition cannot be null");
         }
-        if (expressionAsSelect().getCondition() == null) {
-            expressionAsSelect().setCondition(condition);
+        if (userQueryBuilder.getSelect().getCondition() == null) {
+            userQueryBuilder.getSelect().setCondition(condition);
         } else {
             if (predicate == Predicate.OR) {
-                expressionAsSelect().setCondition(or(expressionAsSelect().getCondition(), condition));
+                userQueryBuilder.getSelect().setCondition(or(userQueryBuilder.getSelect().getCondition(), condition));
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java]: In PR `https://github.com/Talend/tmdm-server-se/pull/747/files`, there is code change in `UnaryLogicOperator.java` for AND/OR operator, does this also need this kind of change? @YohannTalend , @hw-zhu

Comment (hw-zhu) [reply org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java]: PR 747 only modified unary operation NOT, don't involve above snippet of code.

File: org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java (hunk 2 above)
Comment (YohannTalend) [reply org.talend.mdm.query/src/com/amalto/core/query/user/UserQueryBuilder.java]: Done on UserQueryBuilder.java ans StorageQueryTest.java

**Pull request merged**